duration.go

```diff
@@ -122,20 +122,79 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
+// Format formats the given duration into a ISO 8601 duration string (e.g. P1DT6H5M)
+func Format(d time.Duration) string {
+	neg := false
+	if d < 0 {
+		neg = true
+		d = -d
+	}
+
+	if d == 0 {
+		return "PT0S"
+	}
+
+	duration := &Duration{}
+	if d.Hours() >= 8760 {
+		duration.Years = math.Floor(d.Hours() / 8760)
+		d -= time.Duration(duration.Years) * time.Hour * 8760
+	}
+	if d.Hours() >= 730 {
+		duration.Months = math.Floor(d.Hours() / 730)
+		d -= time.Duration(duration.Months) * time.Hour * 730
+	}
+	if d.Hours() >= 168 {
+		duration.Weeks = math.Floor(d.Hours() / 168)
+		d -= time.Duration(duration.Weeks) * time.Hour * 168
+	}
+	if d.Hours() >= 24 {
+		duration.Days = math.Floor(d.Hours() / 24)
+		d -= time.Duration(duration.Days) * time.Hour * 24
+	}
+	if d.Hours() >= 1 {
+		duration.Hours = math.Floor(d.Hours())
+		d -= time.Duration(duration.Hours) * time.Hour
+	}
+	if d.Minutes() >= 1 {
+		duration.Minutes = math.Floor(d.Minutes())
+		d -= time.Duration(duration.Minutes) * time.Minute
+	}
+	duration.Seconds = d.Seconds()
+
+	if neg {
+		return "-" + duration.String()
+	}
+	return duration.String()
+}
+
 // ToTimeDuration converts the *Duration to the standard library's time.Duration
 // note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
 // since obviously those things vary month to month and year to year
 // I used the values that Google's search provided me with as I couldn't find anything concrete on what they should be
 func (duration *Duration) ToTimeDuration() time.Duration {
 	var timeDuration time.Duration
 
-	timeDuration += time.Duration(math.Round(duration.Years * 3.154e+16))
-	timeDuration += time.Duration(math.Round(duration.Months * 2.628e+15))
-	timeDuration += time.Duration(math.Round(duration.Weeks * 6.048e+14))
-	timeDuration += time.Duration(math.Round(duration.Days * 8.64e+13))
-	timeDuration += time.Duration(math.Round(duration.Hours * 3.6e+12))
-	timeDuration += time.Duration(math.Round(duration.Minutes * 6e+10))
-	timeDuration += time.Duration(math.Round(duration.Seconds * 1e+9))
+	if duration.Years != 0 {
```

Comment:
What are these if checks for? If `duration.Years` is 0 then timeDuration isn't changed at all.

Reply:
It is slightly more performant with the checks:
```
pkg: github.com/sosodev/duration
cpu: AMD Ryzen 7 5800X 8-Core Processor
Benchmark_ToTimeDuration (PT1S)                                      355179433    3.383 ns/op
BenchmarkDuration_ToTimeDurationWithoutChecks (PT1S)                 196706413    5.968 ns/op
Benchmark_ToTimeDuration (PT12H30M)                                  271454889    4.409 ns/op
BenchmarkDuration_ToTimeDurationWithoutChecks (PT12H30M)             177455833    6.787 ns/op
Benchmark_ToTimeDuration (P3Y6M4DT12H30M5.5S)                        157064444    7.555 ns/op
BenchmarkDuration_ToTimeDurationWithoutChecks (P3Y6M4DT12H30M5.5S)   162975475    7.409 ns/op
```

Reply:
The performance gain you're seeing here might not be consistent. Some CPU might be slower while others are faster. Are you in a situation where you need this library to be highly performant? I'd prefer to drop the checks just to keep the code simple.

Reply:
Not strictly, but faster is always better :)
But in my app, usually, there are only minutes in duration, so it seems unnecessary to multiply and round up zeroes.
I run the test on multiple hosts and for a value `PT5M` the performance is about two times faster with the checks present.

Reply:
Faster is not always better. Optimization without reason is the root of a lot of code quality issues. However, if you explain that these checks are for performance gains in a comment then I'm happy to keep them.